fga/store/cachedFgaTenantStore.go

```diff
@@ -0,0 +1,96 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/golang-lru/v2/expirable"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/status"
+)
+
+type FgaTenantStore interface {
+	GetStoreIDForTenant(ctx context.Context, conn openfgav1.OpenFGAServiceClient, tenantID string) (string, error)
+	GetModelIDForTenant(ctx context.Context, conn openfgav1.OpenFGAServiceClient, tenantID string) (string, error)
+	IsDuplicateWriteError(err error) bool
+	GetCache() *expirable.LRU[string, string]
+}
+
+type Service struct {
+	cache *expirable.LRU[string, string]
+	FgaTenantStore
```

Comment:
```suggestion
```

Reply:
Instead of having it embedded here, consider the compile ttime check like:

```go
var _ FGAStoreHelper = (*Service)(nil)
```

Reply:
Done, although not sure what this line does.